Remove throttler hack

mihails-strasuns opened the pull request.

Commit:
Remove throttling hook hack

Now that throttling hook is run from epoll cycle context, it is not
necessary anymore to handle task termination hooks in any special way.

gavin-norman-sociomantic commented on src/ocean/task/Scheduler.d:
```diff
@@ -555,11 +555,18 @@ final class Scheduler : IScheduler
         }
-        while (this.fiber_pool.queued_tasks.length && this.cycle_pending_task_count);
+        while ((this.fiber_pool.queued_tasks.length ||
+            this.cycle_pending_task_count) && this.state != State.Shutdown);
```
Does this change in the scheduler relate to "Make throttling hook run once per epoll cycle"? (Is this a separate change?)

mihails-strasuns replied:
Related but separate bug fix. Will split.

Commit:
Fix flawed scheduler loop condition

Was modified during epoll cycle introduction and had two flaws:

1) Not accounting for shutdown state
2) Using && instead of ||

Commit:
Run throttling hook once per epoll cycle

matthias-wende-sociomantic reviewed (commented):
Looks good! I'm happy to approve/merge once you're finished :)

gavin-norman-sociomantic commented:
(Approved apart from the failed tests, of course.)

Commit:
Pipe -debug build logs to /dev/null

It hits output length limit in Travis and we only care that it still
compiles.
